arch/arc/core/mpu/Kconfig

```diff
@@ -0,0 +1,38 @@
+# Kconfig - Memory Protection Unit (MPU) configuration options
+
+#
+# Copyright (c) 2017 Synopsys
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+config ARC_MPU_VER
+	int
+	prompt "ARC MPU version"
+	range 2 4
+	default 2
+	help
+	ARC MPU has several versions. For MPU v2, the minimum region is 2048 bytes;
+	For MPU v3, the minimum region is 32 bytes
+
+config ARC_CORE_MPU
+	bool "ARC Core MPU functionalities"
+	depends on CPU_HAS_MPU
+	select THREAD_STACK_INFO
+	default n
+	help
+	  ARC Core MPU functionalities
+
+config MPU_STACK_GUARD
+	bool "Thread Stack Guards"
+	depends on ARC_CORE_MPU
+	default n
+	help
+	  Enable Thread Stack Guards via MPU
```

Comment:
For stack guard,  ARC has more effective way,  i.e., STACK_CHECKING.  But CONFIG_MPU_STACK_GUARD is added here to test the MPU feature.  After I search the zephyr's codes,  it seems MPU_STACK_GUARD is the only place to show MPU's protection.  In the future, it's better to use STACK_CHECKING to do stack protection.